src/resources/views/crud/fields/select2_from_ajax.blade.php

```diff
@@ -212,10 +212,25 @@ function bpFieldInitSelect2FromAjaxElement(element) {
         // when one of those dependencies changes value
         // reset the select2 value
         for (var i=0; i < $dependencies.length; i++) {
-            $dependency = $dependencies[i];
-            $('input[name='+$dependency+'], select[name='+$dependency+'], checkbox[name='+$dependency+'], radio[name='+$dependency+'], textarea[name='+$dependency+']').change(function () {
-                element.val(null).trigger("change");
-            });
+            //if element has name it means is not in repeatable, because in repeatable we strip the names out.
+            if(typeof element.attr('name') != 'undefined') {
+                $(`[name="${$dependency}"], [name="${$dependency}[]"]`).change(function (el) {
+                    //check if we are dealing with inputs as it's expected and not with some random element in page that happen to have prop `name`
+                    if(el.target.nodeName === 'INPUT' || el.target.nodeName === 'SELECT' || el.target.nodeName === 'TEXTAREA') {
+                        $(element.find('option:not([value=""])')).remove();
+                        element.val(null).trigger("change");
+                    }
+                });
```

Comment:
@pxpm if we use the form to find the dependency elements, we can avoid that extra validation of `nodeName === 'INPUT/SELECT/TEXTAREA'`. 

We would be inside the form, it's guaranteed no other field could have same name. What do you think?

```suggestion
                form.find(`[name="${$dependency}"], [name="${$dependency}[]"]`).change(function(el) {
                    $(element.find('option:not([value=""])')).remove();
                    element.val(null).trigger("change");
                });
```

This applies for the other fields too.

Reply:
Agree @promatik 

going to refactor accordingly.